Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -66,8 +66,12 @@
 
 	# Add here commands to install the package into debian/cwiid.
 	$(MAKE) install prefix=$(CURDIR)/debian/cwiid/usr
+        #maemo stuff
+	mkdir -p $(CURDIR)/debian/cwiid/usr/share/applications/hildon
+	cp -v $(CURDIR)/wmgui.desktop $(CURDIR)/debian/cwiid/usr/share/applications/hildon/
 
 
+
 # Build architecture-independent files here.
 binary-indep: build install
 # We have nothing to do by default.
@@ -76,10 +80,10 @@
 binary-arch: build install
 	dh_testdir
 	dh_testroot
-	dh_installchangelogs ChangeLog
-	dh_installdocs
-	dh_installexamples
-#	dh_install
+#	dh_installchangelogs ChangeLog
+#	dh_installdocs
+#	dh_installexamples
+	dh_install --sourcedir=debian/tmp
 #	dh_installmenu
 #	dh_installdebconf	
 #	dh_installlogrotate
@@ -89,7 +93,7 @@
 #	dh_installinit
 #	dh_installcron
 #	dh_installinfo
-	dh_installman
+#	dh_installman
 	dh_link
 	dh_strip
 	dh_compress
Index: wmgui.desktop
===================================================================
--- wmgui.desktop	(revision 0)
+++ wmgui.desktop	(revision 0)
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Version=0.9.2
+Name=WIIRemoteGUI
+Type=Application
+Comment=wiiremote gui
+Exec=wmgui
+TryExec=wmgui
+Path=
+Icon=xmoto.xpm
+Categories=Extra
