Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -24,6 +24,9 @@
 configure-stamp:
 	dh_testdir
 	# Add here commands to configure the package.
+#	intltoolize --copy --automake --force
+	autoreconf -f -i -s
+	CFLAGS="$(CFLAGS)" ./configure --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE) --prefix=/usr --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info
 
 	touch configure-stamp
 
