Index: debian/changelog
===================================================================
--- debian/changelog	(revision 1)
+++ debian/changelog	(working copy)
@@ -1,3 +1,9 @@
+libgpewidget (0.114-mud1) unstable; urgency=low
+
+  * svn revision 8911
+
+ -- Luca Ognibene <luca.ognibene@gmail.com>  Thu,  8 Feb 2007 14:40:21 +0100
+
 libgpewidget (0.73-1) unstable; urgency=low
 
   * New upstream version.
Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -31,6 +31,8 @@
 	# Add here commands to configure the package.
 
+	sh autogen.sh
+	./configure --prefix=/usr --enable-hildon --disable-cairo
 	touch configure-stamp
 
 
 build: build-stamp
@@ -61,7 +63,7 @@
 
 	# Add here commands to install the package into debian/tmp
 	$(MAKE) install PREFIX=/usr DESTDIR=$(CURDIR)/debian/tmp
-	$(MAKE) install-devel PREFIX=/usr DESTDIR=$(CURDIR)/debian/tmp
+	#$(MAKE) install-devel PREFIX=/usr DESTDIR=$(CURDIR)/debian/tmp
 
 
 # Build architecture-independent files here.
Index: configure.ac
===================================================================
--- configure.ac	(revision 1)
+++ configure.ac	(working copy)
@@ -36,7 +36,7 @@
 	ENABLE_CAIRO=$enableval
 ],
 [
-       	PKG_CHECK_EXISTS(cairo, [ENABLE_CAIRO=yes], [ENABLE_CAIRO=no])
+ 	ENABLE_CAIRO=false
 ])
 
 ENABLE_HILDON=false
