Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -30,6 +30,10 @@
 configure-stamp:
 	dh_testdir
 	# Add here commands to configure the package.
+	intltoolize --copy --automake --force
+	echo Work round scratchbox bug with aclocal search ordering
+	ACLOCAL="`which aclocal` -I /usr/share/aclocal"  autoreconf -f -i -s
+	./configure --enable-hildon --disable-cairo --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE) --prefix=/usr --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info
 
 	touch configure-stamp
 
@@ -61,7 +65,7 @@
 
 	# Add here commands to install the package into debian/tmp
 	$(MAKE) install PREFIX=/usr DESTDIR=$(CURDIR)/debian/tmp
-	$(MAKE) install-devel PREFIX=/usr DESTDIR=$(CURDIR)/debian/tmp
+#	$(MAKE) install-devel PREFIX=/usr DESTDIR=$(CURDIR)/debian/tmp
 
 
 # Build architecture-independent files here.
@@ -73,15 +77,15 @@
 	dh_testdir
 	dh_testroot
 	dh_movefiles
-
-	dh_installdocs
-	dh_installman
-	dh_installchangelogs 
+#
+#	dh_installdocs
+#	dh_installman
+#	dh_installchangelogs 
 	dh_link
 	dh_strip
 	dh_compress
 	dh_fixperms
-	dh_makeshlibs
+	dh_makeshlibs -V
 	dh_installdeb
 	dh_shlibdeps -V
 	dh_gencontrol
Index: configure.ac
===================================================================
--- configure.ac	(revision 1)
+++ configure.ac	(working copy)
@@ -25,7 +25,7 @@
 PKG_PROG_PKG_CONFIG
 
 #gtk-doc for documentation
-GTK_DOC_CHECK(1.2)
+#GTK_DOC_CHECK(1.2)
 
 # library dependencies
 DEPS="gtk+-2.0 >= 2.4 glib-2.0 >= 2.6"
@@ -36,7 +36,7 @@
 	ENABLE_CAIRO=$enableval
 ],
 [
-       	PKG_CHECK_EXISTS(cairo, [ENABLE_CAIRO=yes], [ENABLE_CAIRO=no])
+ 	ENABLE_CAIRO=false
 ])
 
 ENABLE_HILDON=false
@@ -69,7 +69,7 @@
 
 AC_CONFIG_FILES([Makefile libgpewidget.pc 
                  libgpewidget-uninstalled.pc 
-				 doc/Makefile po/Makefile.in])
+				 po/Makefile.in])
 AC_OUTPUT
 
 echo "
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 1)
+++ Makefile.am	(working copy)
@@ -1,5 +1,5 @@
 
-SUBDIRS = doc po
+SUBDIRS = po
 
 if HILDON
 
