Index: debian/control
===================================================================
--- debian/control	(revision 1)
+++ debian/control	(working copy)
@@ -81,17 +81,6 @@
  This package contains a commandline interface to the GNU TLS library,
  which can be used to set up secure connections from e.g. shell scripts.
 
-Package: gnutls-doc
-Priority: optional
-Architecture: all
-Section: doc
-Depends: ${misc:Depends}
-Description: the GNU TLS library - documentation and examples
- gnutls is a portable library which implements the Transport Layer
- Security (TLS) 1.0 and Secure Sockets Layer (SSL) 3.0 protocols.
- .
- This package contains all the gnutls documentation.
-
 Package: libgnutlsxx13
 Priority: optional
 Architecture: any
Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -9,8 +9,8 @@
 include /usr/share/cdbs/1/class/autotools.mk
 include /usr/share/cdbs/1/rules/simple-patchsys.mk
 
-DEB_CONFIGURE_EXTRA_FLAGS = --with-mcrypt=no --enable-ld-version-script
-DEB_MAKE_CHECK_TARGET = check
+DEB_CONFIGURE_EXTRA_FLAGS = --with-mcrypt=no --enable-ld-version-script --disable-gtk-doc
+#DEB_MAKE_CHECK_TARGET = check
 DEB_DH_STRIP_ARGS = --dbg-package=libgnutls13-dbg
 DEB_DH_MAKESHLIBS_ARGS_libgnutls13 := -V 'libgnutls13 (>=1.6.3-0)'
 DEB_COMPRESS_EXCLUDE := gnutls.pdf
Index: Makefile.in
===================================================================
--- Makefile.in	(revision 1)
+++ Makefile.in	(working copy)
@@ -349,8 +349,8 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
-SUBDIRS = gl lgl includes lib libextra src doc tests po
+DISTCHECK_CONFIGURE_FLAGS = --disable-gtk-doc
+SUBDIRS = gl lgl includes lib libextra src
 EXTRA_DIST = mkinstalldirs
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4 -I lgl/m4
 tag = $(PACKAGE)_`echo $(VERSION) | sed 's/\./_/g'`
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 1)
+++ Makefile.am	(working copy)
@@ -19,9 +19,9 @@
 # along with this file; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
-DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
+DISTCHECK_CONFIGURE_FLAGS = --disable-gtk-doc
 
-SUBDIRS = gl lgl includes lib libextra src doc tests po
+SUBDIRS = gl lgl includes lib libextra src
 
 EXTRA_DIST = mkinstalldirs
 
