Index: debian/control
===================================================================
--- debian/control	(revision 1)
+++ debian/control	(working copy)
@@ -73,30 +73,4 @@
  .
  This package contains the shared library.
 
-Package: libsoup2.2-doc
-Section: devel
-Architecture: all
-Depends: ${misc:Depends}
-Suggests: libsoup2.2-dev
-Description: an HTTP library implementation in C -- API Reference
- It was originally part of a SOAP (Simple Object Access Protocol)
- implementation called Soup, but the SOAP and non-SOAP parts have now been
- split into separate packages.
- .
- libsoup uses the Glib main loop and is designed to work well with GTK
- applications. This enables GNOME applications to access HTTP servers
- on the network in a completely asynchronous fashion, very similar to
- the Gtk+ programming model (a synchronous operation mode is also
- supported for those who want it).
- .
- Features:
-  * Both asynchronous (GMainLoop and callback-based) and synchronous APIs
-  * Automatically caches connections
-  * SSL Support using GnuTLS
-  * Proxy support, including authentication and SSL tunneling
-  * Client support for Digest, NTLM, and Basic authentication
-  * Server support for Digest and Basic authentication
-  * Basic client-side SOAP and XML-RPC support
- .
- This package contains the documentation.
 
Index: debian/control.in
===================================================================
--- debian/control.in	(revision 1)
+++ debian/control.in	(working copy)
@@ -73,30 +73,4 @@
  .
  This package contains the shared library.
 
-Package: libsoup2.2-doc
-Section: devel
-Architecture: all
-Depends: ${misc:Depends}
-Suggests: libsoup2.2-dev
-Description: an HTTP library implementation in C -- API Reference
- It was originally part of a SOAP (Simple Object Access Protocol)
- implementation called Soup, but the SOAP and non-SOAP parts have now been
- split into separate packages.
- .
- libsoup uses the Glib main loop and is designed to work well with GTK
- applications. This enables GNOME applications to access HTTP servers
- on the network in a completely asynchronous fashion, very similar to
- the Gtk+ programming model (a synchronous operation mode is also
- supported for those who want it).
- .
- Features:
-  * Both asynchronous (GMainLoop and callback-based) and synchronous APIs
-  * Automatically caches connections
-  * SSL Support using GnuTLS
-  * Proxy support, including authentication and SSL tunneling
-  * Client support for Digest, NTLM, and Basic authentication
-  * Server support for Digest and Basic authentication
-  * Basic client-side SOAP and XML-RPC support
- .
- This package contains the documentation.
 
Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -17,7 +17,7 @@
 
 DEB_SHLIBDEPS_INCLUDE_libsoup2.2-dev += debian/libsoup2.2-${SOUP_SOVERSION}/usr/lib
 DEB_DH_MAKESHLIBS_ARGS_ALL += -V'$(SHLIBS)'
-DEB_CONFIGURE_EXTRA_FLAGS += --enable-gtk-doc
+DEB_CONFIGURE_EXTRA_FLAGS += --disable-gtk-doc
 
 install/libsoup2.2-doc::
 	# rename the devhelp files to match the parent directory as otherwise
