Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -46,7 +46,7 @@
 install: build
 	dh_testdir
 	dh_testroot
-	dh_prep
+	dh_clean -k
 	dh_installdirs
 
 	# Add here commands to install the package into debian/proxytunnel.
@@ -62,7 +62,7 @@
 	dh_testdir
 	dh_testroot
 #	dh_installdebconf	
-	dh_installdocs
+#	dh_installdocs
 #	dh_installexamples
 #	dh_installmenu
 #	dh_installlogrotate
@@ -74,7 +74,7 @@
 #	dh_installman
 #	dh_installinfo
 #	dh_undocumented
-	dh_installchangelogs CHANGES
+#	dh_installchangelogs CHANGES
 	dh_link
 	dh_strip
 	dh_compress
@@ -84,6 +84,7 @@
 #	dh_perl
 	dh_shlibdeps
 	dh_gencontrol
+	if which maemo-optify >/dev/null ; then maemo-optify ; fi
 	dh_md5sums
 	dh_builddeb
 
