Index: debian/postinst
===================================================================
--- debian/postinst	(revision 0)
+++ debian/postinst	(revision 0)
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+gtk-update-icon-cache -f /usr/share/icons/hicolor
Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -15,7 +15,7 @@
 DEB_HOST_GNU_TYPE   ?= $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
 DEB_BUILD_GNU_TYPE  ?= $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
 
-
+ICON_DIR = usr/share/icons/hicolor
 CFLAGS = -Wall -g
 
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
@@ -27,7 +27,7 @@
 config.status: configure
 	dh_testdir
 	# Add here commands to configure the package.
-	./configure --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE) --prefix=/usr --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info CFLAGS="$(CFLAGS)" LDFLAGS="-Wl,-z,defs"
+	./configure --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE) --prefix=/usr --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info CFLAGS="$(CFLAGS)" LDFLAGS="-Wl,-z,defs" --disable-darwin --disable-xsmp --disable-netbeans --enable-gui=no --disable-gtktest --disable-gpm --disable-acl --disable-nls --disable-darwin --disable-xsmp --disable-netbeans --enable-gui=no --disable-gtktest --disable-gpm --disable-acl --disable-nls --enable-multibyte --disable-darwin --disable-xsmp --disable-netbeans --enable-gui=no --disable-gtktest --disable-gpm --disable-acl --disable-nls --enable-multibyte --disable-darwin --disable-xsmp --disable-netbeans --enable-gui=no --disable-gtktest --disable-gpm --disable-acl --disable-nls --enable-multibyte --disable-darwin --disable-xsmp --disable-netbeans --enable-gui=no --disable-gtktest --disable-gpm --disable-acl --disable-nls --enable-multibyte
 
 
 build: build-stamp
@@ -63,11 +63,15 @@
 	dh_testroot
 	dh_clean -k 
 	dh_installdirs
-
+	
 	# Add here commands to install the package into debian/vim.
 	$(MAKE) prefix=$(CURDIR)/debian/vim/usr install
+	install -D --mode=0644 $(MUD_PACKAGES_RELDIR)/icon/vim-26.png debian/vim/$(ICON_DIR)/26x26/hildon/vim.png
+	install -D --mode=0644 $(MUD_PACKAGES_RELDIR)/icon/vim-40.png debian/vim/$(ICON_DIR)/40x40/hildon/vim.png
+	install -D --mode=0644 $(MUD_PACKAGES_RELDIR)/icon/vim-64.png debian/vim/$(ICON_DIR)/scalable/hildon/vim.png
+	install -D --mode=0755 $(MUD_PACKAGES_RELDIR)/extras/vim.launch debian/vim/usr/bin/vim.launch
+	install -D --mode=0644 $(MUD_PACKAGES_RELDIR)/extras/vim.desktop debian/vim/usr/share/applications/hildon/vim.desktop
 
-
 # Build architecture-independent files here.
 binary-indep: build install
 # We have nothing to do by default.
@@ -76,11 +80,11 @@
 binary-arch: build install
 	dh_testdir
 	dh_testroot
-	dh_installchangelogs 
-	dh_installdocs
-	dh_installexamples
-#	dh_install
-#	dh_installmenu
+#	dh_installchangelogs 
+#	dh_installdocs
+#	dh_installexamples
+	dh_install --sourcedir=debian/tmp
+	dh_installmenu
 #	dh_installdebconf	
 #	dh_installlogrotate
 #	dh_installemacsen
@@ -90,7 +94,7 @@
 #	dh_installinit
 #	dh_installcron
 #	dh_installinfo
-	dh_installman
+#	dh_installman
 	dh_link
 	dh_strip
 	dh_compress
