Index: debian/control
===================================================================
--- debian/control	(revision 1)
+++ debian/control	(working copy)
@@ -15,5 +15,8 @@
 Section: libs
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends}
+Provides: libmimedir-gnome1
+Conflicts: libmimedir-gnome1
+Replaces: libmimedir-gnome1
 Description: <insert up to 60 chars description>
  <insert long description, indented with spaces>
Index: debian/libmimedir-gnome0.dirs
===================================================================
--- debian/libmimedir-gnome0.dirs	(revision 0)
+++ debian/libmimedir-gnome0.dirs	(revision 0)
@@ -0,0 +1 @@
+usr/lib
Index: debian/rules
===================================================================
--- debian/rules	(revision 1)
+++ debian/rules	(working copy)
@@ -33,6 +33,7 @@
 configure-stamp:
 	dh_testdir
 	# Add here commands to configure the package.
+	REQUIRED_AUTOMAKE_VERSION=1.5 ./autogen.sh --enable-hildon --disable-cairo --disable-gtk-doc --host=$(DEB_HOST_GNU_TYPE) --build=$(DEB_BUILD_GNU_TYPE) --prefix=/usr --mandir=\$${prefix}/share/man --infodir=\$${prefix}/share/info
 
 	touch configure-stamp
 
@@ -74,10 +75,10 @@
 binary-arch: build install
 	dh_testdir
 	dh_testroot
-	dh_installchangelogs ChangeLog
-	dh_installdocs
-	dh_installexamples
-#	dh_install
+#	dh_installchangelogs ChangeLog
+#	dh_installdocs
+#	dh_installexamples
+	dh_install --sourcedir=debian/tmp
 #	dh_installmenu
 #	dh_installdebconf	
 #	dh_installlogrotate
@@ -87,14 +88,14 @@
 #	dh_installinit
 #	dh_installcron
 #	dh_installinfo
-	dh_installman
+#	dh_installman
 	dh_link
 	dh_strip
 	dh_compress
 	dh_fixperms
 #	dh_perl
 #	dh_python
-#	dh_makeshlibs
+	dh_makeshlibs -V
 	dh_installdeb
 	dh_shlibdeps
 	dh_gencontrol
Index: debian/libmimedir-gnome0.install
===================================================================
--- debian/libmimedir-gnome0.install	(revision 0)
+++ debian/libmimedir-gnome0.install	(revision 0)
@@ -0,0 +1 @@
+usr/lib/lib*.so.*
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 1)
+++ Makefile.am	(working copy)
@@ -1,4 +1,4 @@
-SUBDIRS = mimedir bin po docs
+SUBDIRS = mimedir bin 
 
 EXTRA_DIST = \
 	COPYING.LIB		\
