Index: debian/control
===================================================================
--- debian/control	(revision 1)
+++ debian/control	(working copy)
@@ -2,11 +2,11 @@
 Section: unknown
 Priority: optional
 Maintainer: unknown <>
-Build-Depends: debhelper (>= 4.0.0)
+Build-Depends: debhelper (>= 4.0.0) , libsdl-gfx-dev
 Standards-Version: 3.6.1
 
 Package: xmoto
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}
+Depends: ${shlibs:Depends}, ${misc:Depends}, lua , libsdl-gfx1
 Description: <insert up to 60 chars description>
  <insert long description, indented with spaces>
